ingestion/src/metadata/utils/secrets/local_secrets_manager.py

```diff
@@ -0,0 +1,73 @@
+#  Copyright 2022 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Secrets manager implementation for local secrets manager
+"""
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+    SecretsManagerProvider,
+)
+from metadata.generated.schema.entity.services.connections.serviceConnection import (
+    ServiceConnection,
+)
+from metadata.generated.schema.metadataIngestion.workflow import SourceConfig
+from metadata.utils.secrets.secrets_manager import (
+    SecretsManager,
+    ServiceWithConnectionType,
+    logger,
+)
+
+
+class LocalSecretsManager(SecretsManager):
+    """
+    LocalSecretsManager is used when there is not a secrets' manager configured.
+    """
+
+    provider: str = SecretsManagerProvider.local.name
+
+    def add_auth_provider_security_config(
+        self, open_metadata_connection: OpenMetadataConnection
+    ) -> None:
+        """
+        The LocalSecretsManager does not modify the OpenMetadataConnection object
+        """
+        logger.debug(
+            f"Adding auth provider security config using {self.provider} secrets' manager"
+        )
+        return None
+
+    def retrieve_service_connection(
+        self,
+        service: ServiceWithConnectionType,
+        service_type: str,
+    ) -> ServiceConnection:
+        """
+        The LocalSecretsManager does not modify the ServiceConnection object
+        """
+        logger.debug(
+            f"Retrieving service connection from {self.provider} secrets' manager for {service_type} - {service.name}"
+        )
+        return ServiceConnection(__root__=service.connection)
```

Comment:
Is it needed to pass the `__root__`? Here you might be good to go returning just `service.connection`

Reply:
`service.connection` is an object that could be `DashboardConnection`, `DatabaseConnection`, etc. So in this way, what I am returning is the generic service connection. `Source.serviceConnection` is expecting a `ServiceConnection`.

Reply:
cool thanks for explaining, LGTM